### DISCUSSION TITLE
Support for SCAM Pattern

### DISCUSSION THREAD

**Discussion by tayambamwanza** [cyrilletuzi/vscode-angular-schematics#494] [Ideas]
I learned about the SCAM pattern a couple of days ago, would you consider having it as an option for the component generator?

https://dev.to/this-is-angular/emulating-tree-shakable-components-using-single-component-angular-modules-13do

**Comment by cyrilletuzi** [comment]
> would you consider having it as an option for the component generator?

As a reminder, this extension is just providing a graphical interface for existing schematics.

If a community schematic was created for this pattern, the extension would already support it right away, as for all schematics. We would just have to add it to the defaults list so it is automatically detected.

**Reply by tayambamwanza** [reply]
Sounds good to me, there is a schematic from ngneat, I will test the intergration, but if it could be added to the default list that would be great too 💯.

https://github.com/ngneat/scam

**Comment by cyrilletuzi** [comment]
Done in #502, released in v4.13.

**Reply by tayambamwanza** [reply]
You are the bomb dot com 💣

**Comment by cyrilletuzi** [comment]
~~For everyone information, a better SCAM / AIM support has been added to [Schematics Pro](https://cyrilletuzi.gumroad.com/l/schematicspro/).~~

With standalone components, the SCAM pattern is now useless.
